src/libs/Performance.tsx

```diff
@@ -66,204 +29,237 @@ function diffObject(object: Record<string, unknown>, base: Record<string, unknow
+}
+
+/**
+ * Measures the TTI time. To be called when the app is considered to be interactive.
+ */
+function measureTTI(endMark?: string): void {
+    // Make sure TTI is captured when the app is really usable
+    InteractionManager.runAfterInteractions(() => {
+        requestAnimationFrame(() => {
+            measureFailSafe('TTI', 'nativeLaunchStart', endMark);
 
-if (Metrics.canCapturePerformanceMetrics()) {
-    const perfModule = require<ReactNativePerformance>('react-native-performance');
-    perfModule.setResourceLoggingEnabled(true);
-    rnPerformance = perfModule.default;
-
-    Performance.measureFailSafe = (measureName: string, startOrMeasureOptions: string, endMark?: string) => {
-        try {
-            rnPerformance.measure(measureName, startOrMeasureOptions, endMark);
-        } catch (error) {
-            // Sometimes there might be no start mark recorded and the measure will fail with an error
-            if (error instanceof Error) {
-                console.debug(error.message);
+            // we don't want the alert to show on an e2e test session
+            if (!isE2ETestSession()) {
+                printPerformanceMetrics();
             }
+        });
+    });
+}
+
+/*
+ * Monitor native marks that we want to put on the timeline
+ */
+
+const nativeMarksObserver = new PerformanceObserver((list, _observer) => {
+    list.getEntries().forEach((entry: PerformanceEntry) => {
+        if (entry.name === 'nativeLaunchEnd') {
+            measureFailSafe('nativeLaunch', 'nativeLaunchStart', 'nativeLaunchEnd');
+        }
+        if (entry.name === 'downloadEnd') {
+            measureFailSafe('jsBundleDownload', 'downloadStart', 'downloadEnd');
+        }
+        if (entry.name === 'runJsBundleEnd') {
+            measureFailSafe('runJsBundle', 'runJsBundleStart', 'runJsBundleEnd');
+        }
+        if (entry.name === 'appCreationEnd') {
+            measureFailSafe('appCreation', 'appCreationStart', 'appCreationEnd');
+            measureFailSafe('nativeLaunchEnd_To_appCreationStart', 'nativeLaunchEnd', 'appCreationStart');
+        }
+        if (entry.name === 'contentAppeared') {
+            measureFailSafe('appCreationEnd_To_contentAppeared', 'appCreationEnd', 'contentAppeared');
         }
-    };
 
-    /**
-     * Measures the TTI time. To be called when the app is considered to be interactive.
-     */
-    Performance.measureTTI = (endMark?: string) => {
-        // Make sure TTI is captured when the app is really usable
-        InteractionManager.runAfterInteractions(() => {
-            requestAnimationFrame(() => {
-                Performance.measureFailSafe('TTI', 'nativeLaunchStart', endMark);
-
-                // we don't want the alert to show on an e2e test session
-                if (!isE2ETestSession()) {
-                    Performance.printPerformanceMetrics();
-                }
-            });
-        });
-    };
+        // We don't need to keep the observer past this point
```

Comment:
> We don't need to keep the observer past this point

Why not? Mind adding the answer to the comment to improve the value of the comment?

Reply:
Good point, updated.